File: src/data/courses/dsa/greedy/interval-based/meeting-rooms-ii/code/python/meeting-rooms-ii.py
def min_meeting_rooms(intervals):
    start = sorted([i[0] for i in intervals])  # Sort start times
    end = sorted([i[1] for i in intervals])  # Sort end times
    rooms, end_ptr = 0, 0
    for i in range(len(intervals)):
        if start[i] < end[end_ptr]:  # Meeting starts before earliest ends
            rooms += 1  # Need new room
        else:
            end_ptr += 1  # Reuse room
    return rooms

File: src/data/courses/dsa/greedy/interval-based/meeting-rooms-ii/code/java/MeetingRoomsIi.java
class MeetingRoomsII {
    public int minMeetingRooms(int[][] intervals) {
        int n = intervals.length;
        int[] start = new int[n], end = new int[n];
        for (int i = 0; i < n; i++) {
            start[i] = intervals[i][0];
            end[i] = intervals[i][1];
        }
        Arrays.sort(start); // Sort start times
        Arrays.sort(end); // Sort end times
        
        int rooms = 0, endPtr = 0;
        for (int i = 0; i < n; i++) {
            if (start[i] < end[endPtr]) { // Meeting starts before earliest ends
                rooms++; // Need new room
            } else {
                endPtr++; // Reuse room
            }
        }
        return rooms;
    }
}